client/landing/stepper/hooks/use-site-copy.tsx

```diff
@@ -0,0 +1,79 @@
+import { WPCOM_FEATURES_COPY_SITE } from '@automattic/calypso-products';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { useDispatch } from '@wordpress/data';
+import { useEffect, useMemo, useCallback } from 'react';
+import { useDispatch as useReduxDispatch, useSelector } from 'react-redux';
+import { useQuerySitePurchases } from 'calypso/components/data/query-site-purchases';
+import { ONBOARD_STORE } from 'calypso/landing/stepper/stores';
+import { clearSignupDestinationCookie } from 'calypso/signup/storageUtils';
+import { getCurrentUserId } from 'calypso/state/current-user/selectors';
+import {
+	hasLoadedSitePurchasesFromServer,
+	isFetchingSitePurchases,
+} from 'calypso/state/purchases/selectors';
+import siteHasFeature from 'calypso/state/selectors/site-has-feature';
+import { fetchSiteFeatures } from 'calypso/state/sites/features/actions';
+import type { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+
+function useSafeSiteHasFeature( siteId: number, feature: string ) {
+	const dispatch = useReduxDispatch();
+	useEffect( () => {
+		dispatch( fetchSiteFeatures( siteId ) );
+	}, [ dispatch, siteId ] );
+
+	return useSelector( ( state ) => {
+		return siteHasFeature( state, siteId, feature );
+	} );
+}
+
+export const useSiteCopy = ( site: SiteExcerptData ) => {
+	const userId = useSelector( ( state ) => getCurrentUserId( state ) );
+	const hasCopySiteFeature = useSafeSiteHasFeature( site.ID, WPCOM_FEATURES_COPY_SITE );
+	const plan = site?.plan;
+	const isSiteOwner = site.site_owner === userId;
+	useQuerySitePurchases( site.ID );
+	const isLoadingPurchase = useSelector(
+		( state ) => isFetchingSitePurchases( state ) || ! hasLoadedSitePurchasesFromServer( state )
+	);
+
+	const { setPlanCartItem } = useDispatch( ONBOARD_STORE );
+
+	const shouldShowSiteCopyItem = useCallback( () => {
+		return hasCopySiteFeature && isSiteOwner && plan && ! isLoadingPurchase;
+	}, [ hasCopySiteFeature, isSiteOwner, plan, isLoadingPurchase ] );
+
+	const startSiteCopy = useCallback(
+		( recordTracks: () => void ) => {
+			if ( ! plan ) {
+				return;
+			}
+			clearSignupDestinationCookie();
+			setPlanCartItem( { product_slug: plan.product_slug } );
+			recordTracks();
+		},
+		[ plan, setPlanCartItem ]
+	);
+
+	return useMemo(
+		() => ( {
+			shouldShowSiteCopyItem,
+			startSiteCopy,
+		} ),
+		[ shouldShowSiteCopyItem, startSiteCopy ]
```

Comment:
Would it make sense if `shouldShowSiteCopyItem` is a value instead of a function?

Reply:
This is what was discussed originally with Daniel. However it makes more sense to me as well, and I was not feeling very well about that being a function. I will change it to a value. Thanks for your comment